tests/_client.lua

```diff
@@ -0,0 +1,33 @@
+local spawn = require("awful.spawn")
+
+-- This file provide a simple, yet flexible, test client.
+-- It is used to test the `awful.rules`
+
+return function(class, title)
+    local cmd = table.concat {
+        "echo \"",
+        "local lgi = require 'lgi';",
+        "local Gtk = lgi.require('Gtk');",
+        "local GdkPixbuf = lgi.GdkPixbuf;",
+        "Gtk.init();",
+        "local class = '",
+        class or 'test_app',"';",
+        "local window = Gtk.Window {",
+        "    default_width  = 100,",
+        "    default_height = 100,",
+        "    title          = '",
+            title or 'Awesome test client',
+        "'};",
+        "window:set_wmclass(class, class);",
+        "local app = Gtk.Application {",
+        "    application_id = 'org.awesomewm.tests.",class,
+        "'};",
+        "function app:on_activate()",
+        "    window.application = self;",
+        "    window:show_all();",
+        "end;",
+        "app:run {''}\" | lua"
```

Comment:
By the way, right now I'm on a Fedora box where `urxvt` does not support startup-notification and thus the sn-related tests always fail. I *think* that GTK automatically supports startup-notification and so this test-app could be used instead of `urxvt` in the tests.

Remind me to actually implement this eventually...